sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/controller/group/global/ThroughputControlContainerManager.java

```diff
@@ -0,0 +1,185 @@
+    private ThroughputGlobalControlClientItem clientItem;
+
+    public ThroughputControlContainerManager(ThroughputGlobalControlGroup group) {
+        checkNotNull(group, "Global control group config can not be null");
+
+        this.globalControlContainer = group.getGlobalControlContainer();
+        this.group = group;
+
+        this.clientItemId = this.group.getId()  + CLINT_ITEM_ID_SUFFIX;
+        this.clientItemPartitionKeyValue = this.group.getGroupName() + CLIENT_ITEM_PARTITION_KEY_VALUE_SUFFIX;
+        this.configItemId = this.group.getId() + CONFIG_ITEM_ID_SUFFIX;
+        this.configItemPartitionKeyValue = this.group.getGroupName() + CONFIG_ITEM_PARTITION_KEY_VALUE_SUFFIX;
+    }
+
+    public Mono<ThroughputGlobalControlClientItem> createGroupClientItem(double loadFactor) {
+        return Mono.just(new ThroughputGlobalControlClientItem(
+                                this.clientItemId, this.clientItemPartitionKeyValue, loadFactor, this.group.getControlItemExpireInterval()))
+            .flatMap(groupClientItem -> this.globalControlContainer.createItem(groupClientItem))
+            .flatMap(itemResponse -> {
+                this.clientItem = itemResponse.getItem();
+                return Mono.just(this.clientItem);
+            });
+    }
+
+    /**
+     * Get or create the throughput global control config item.
+     * This is to make sure all the clients are using the same configuration for the group.
+     *
+     * The config item in the control container will be used as the source of truth.
+     * If the client has a different config, it will be overwritten by the one in the control container.
+     *
+     * @return A {@link ThroughputGlobalControlClientItem}.
+     */
+    public Mono<ThroughputGlobalControlConfigItem> getOrCreateConfigItem() {
+        ThroughputGlobalControlConfigItem expectedConfigItem =
+            new ThroughputGlobalControlConfigItem(
+                this.configItemId,
+                this.configItemPartitionKeyValue,
+                this.group.getTargetThroughput(),
+                this.group.getTargetThroughputThreshold(),
+                this.group.isDefault());
+
+        return this.globalControlContainer.readItem(
+                    this.configItemId,
+                    new PartitionKey(this.configItemPartitionKeyValue),
+                    ThroughputGlobalControlConfigItem.class)
+            .onErrorResume(throwable -> {
+                CosmosException cosmosException = Utils.as(Exceptions.unwrap(throwable), CosmosException.class);
+                if (cosmosException != null && cosmosException.getStatusCode() == HttpConstants.StatusCodes.NOTFOUND) {
+                    // hooray, you are the first one, needs to create the config file now
+                    return this.globalControlContainer.createItem(expectedConfigItem);
+                }
+
+                return Mono.error(throwable);
+            })
+            .retryWhen(RetrySpec.max(1).filter(throwable -> {
```

Comment:
Don't think 1 is sufficient there - read could return 404 multiple times from different replicas, correct? - If you want to limit it to avoid the pitfall of endless loops etc. use something like 10 or 20 or whatever - reaosnably high static number to give enough room for dirty reads.

Reply:
yea, make sense. will update to 10 in the next iteration.